exampleCourse/questions/demo/annotated/engCircuit/server.py

```diff
@@ -40,33 +48,46 @@ def file(data):
             drawing += elm.Resistor().left().label(R1)
             drawing.pop()
             drawing += elm.BatteryCell().down().label(Vt,'bottom')
-        return drawing.get_imagedata()
+
+    return drawing.get_imagedata()
+
 
 def generate(data):
-    ask = ["equivalent resistance $R_T$", "current from the power supply $I_T$"]
-    which = random.choice([0, 1])
-    data["params"]["ask"] = ask[which]
-    label = ["$R_T$", "$I_T$"]
-    data["params"]["lab"] = label[which]
 
+    # Randomly choose Vt, R1, R2, R3
     Vt = random.randint(100, 200)
     data["params"]["Vt"] = Vt
+
     R1 = random.choice(list(range(20, 180, 10)))
     data["params"]["R1"] = R1
+    
     R2 = random.choice(list(range(20, 180, 20)))
     data["params"]["R2"] = R2
+    
     R3 = random.choice(list(range(20, 100, 5)))
     data["params"]["R3"] = R3
-    unit = ["$\\Omega$", "A"]
-    data["params"]["unit"] = unit[which]
-
+    
+    # Randomly choose variant of the question
+    variant = random.choice([0, 1])
+    match variant:
+        case 0: # Equivalent resistance of 3 parallel resistors
+            data["params"]["ask"] = "equivalent resistance $R_T$"
+            data["params"]["whichfig"] = "R_T"
```

Comment:
Still need to set whichfig independently of `variant`.

Reply:
Done here: https://github.com/PrairieLearn/PrairieLearn/pull/8179/commits/e9d7d5ac1bcf8fca62c502f618e38d30042648db